platform/security/platform-security-core-api/src/main/java/ddf/security/samlp/SamlProtocol.java

```diff
@@ -485,13 +487,21 @@ public static AttributeQuery createAttributeQuery(Issuer issuer, Subject subject
     return createAttributeQuery(issuer, subject, null);
   }
 
-  public static LogoutRequest createLogoutRequest(Issuer issuer, NameID nameId, String id) {
+  public static LogoutRequest createLogoutRequest(
+      Issuer issuer, NameID nameId, String id, List<String> sessionIndexes) {
     LogoutRequest logoutRequest = logoutRequestBuilder.buildObject();
     logoutRequest.setID(id);
     logoutRequest.setIssuer(issuer);
     logoutRequest.setNameID(nameId);
     logoutRequest.setIssueInstant(DateTime.now());
     logoutRequest.setVersion(SAMLVersion.VERSION_20);
+    for (String index : sessionIndexes) {
+      SessionIndexBuilder builder = new SessionIndexBuilder();
```

Comment:
❓ Could we create the `SessionIndexBuilder` only once and reuse it instead of creating a new one each time?

Reply:
Good catch! Fixing it.